### ISSUE TITLE
jsx-a11y/control-has-associated-label throws an error on table elements

### ISSUE THREAD

**Issue by elianarlivingston** [jsx-eslint/eslint-plugin-jsx-a11y#959]
Hello! I'm experiencing an issue with lint on the 'td' or 'th' elements. Is it correct that this error is being generated? Thank you in advance!

<img width="986" alt="Captura de pantalla 2023-11-03 a la(s) 03 52 41" src="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/assets/59533564/fa18b9d7-0eec-48d4-bfc1-3e0c0e3e8e6a">

**Comment by christopher-william** [comment]
I have the same problem, with this version 6.8

**Comment by AlanGreene** [comment]
I've just run into the same issue updating to 6.8.0.

From a quick review of the changelog and recent history it looks like an intentional change, or at the very least a side effect of an intentional change: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/919

In my case it's triggered by an empty `<td>` used in a skeleton loading state.

**Comment by ZeeshanTamboli** [comment]
Same issue for us in Material UI - https://github.com/mui/material-ui/pull/39752.

**Comment by EKarpinsky** [comment]
Hi, is there any update on this?

**Comment by jessebeach** [comment]
Can I call you later?Sent from my Apple WatchOn Jan 8, 2024, at 12:42, EKarpinsky ***@***.***> wrote:
Hi, is there any update on this?

**Comment by frantic1048** [comment]
We are sticking to the older 6.7.1 version to avoid this issue.

Versions 6.8 and the recent [6.9](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/CHANGELOG.md#v690---2024-06-19) still encounter this issue.

**Comment by ljharb** [comment]
@elianarlivingston @christopher-william @frantic1048 can you provide some code that reproduces the issue?

**Comment by frantic1048** [comment]
Here's reproduction repo: https://github.com/frantic1048/eslint-plugin-jsx-a11y-issue-959

- ESLint output: https://github.com/frantic1048/eslint-plugin-jsx-a11y-issue-959/actions/runs/9739279809/job/26874221357
- Code: https://github.com/frantic1048/eslint-plugin-jsx-a11y-issue-959/blob/master/src/App.tsx

It appears that this issue is related to the case where a `<th>` element has only one self-closing child tag.

**Comment by ZeeshanTamboli** [comment]
Shouldn't `th` and `td` be added to `ignoreElements` in the rule [here](https://github.com/airbnb/javascript/blob/c25bce83be4db06e6a221d79686c485cd2ed5d5d/packages/eslint-config-airbnb/rules/react-a11y.js#L77-L85)? Would that fix the issue?

**Comment by ljharb** [comment]
I suppose that'd be one solution.

**Comment by ZeeshanTamboli** [comment]
> I suppose that'd be one solution.

@ljharb I can add that. But will `eslint-config-airbnb` be released? I am hesitant because I see it was released two years ago. We extend `eslint-config-airbnb` which has this a11y rule.

**Comment by ljharb** [comment]
@ZeeshanTamboli the rule is part of this plugin, not the airbnb config, so you don't need the airbnb config to do anything to be able to use new releases of this plugin. that said, yes, the airbnb config will be released for eslint 9 support once everything it depends on has it.

**Comment by ZeeshanTamboli** [comment]
@ljharb I understand the rule is part of this plugin. What I meant is that we extend the Airbnb config (`extends: ['eslint-config-airbnb']`), which uses the rule defined [here](https://github.com/airbnb/javascript/blob/c25bce83be4db06e6a221d79686c485cd2ed5d5d/packages/eslint-config-airbnb/rules/react-a11y.js#L77-L85). If I update the `ignoreElements` list there, wouldn’t `eslint-config-airbnb` need a new release?

**Comment by ljharb** [comment]
You can override that rule config however you like, without a new release.

**Comment by ZeeshanTamboli** [comment]
> You can override that rule config however you like, without a new release.

Yes, but that would be a workaround just to wait for the actual fix to happen.

**Comment by ljharb** [comment]
Yep, that's right

**Comment by ZeeshanTamboli** [comment]
> Yep, that's right

:P So can't we release `eslint-config-airbnb` minor patch?

**Comment by ljharb** [comment]
This wouldn't be the repo to ask on, and this is an issue about this plugin, not about the airbnb config.

**Comment by ZeeshanTamboli** [comment]
> This wouldn't be the repo to ask on, and this is an issue about this plugin, not about the airbnb config.

Agree. Created https://github.com/airbnb/javascript/issues/3069.
